nautobot/utilities/renderers.py

```diff
@@ -0,0 +1,176 @@
+        view = renderer_context["view"]
+        request = renderer_context["request"]
+        instance = view.get_object()
+        model = view.queryset.model
+        form = None
+        table = None
+        form_class = view.get_form_class()
+        content_type = ContentType.objects.get_for_model(model)
+        view.queryset = view.alter_queryset(request)
+        # Compile a dictionary indicating which permissions are available to the current user for this model
+        permissions = self.construct_user_permissions(request, model)
+        # Construct valid actions
+        valid_actions = self.validate_action_buttons(view, request)
+        obj = view.alter_obj_for_edit(instance, request, view.args, view.kwargs)
+        if data.get("form"):
+            form = data["form"]
+        else:
+            if view.action == "list":
+                filter_params = self.get_filter_params(view, request)
+                if view.filterset_class:
+                    filterset = view.filterset_class(filter_params, view.queryset)
+                    view.queryset = filterset.qs
+                    if not filterset.is_valid():
+                        messages.error(
+                            request,
+                            mark_safe(f"Invalid filters were specified: {filterset.errors}"),
+                        )
+                        view.queryset = view.queryset.none()
+                table = self.construct_table(view, request, permissions)
+            elif view.action == "destroy":
+                form = form_class(initial=request.GET)
+            elif view.action in ["create", "update"]:
+                initial_data = normalize_querydict(request.GET)
+                form = form_class(instance=obj, initial=initial_data)
+                restrict_form_fields(form, request.user)
+            elif view.action == "bulk_destroy":
+                if request.POST.get("_all"):
+                    if view.filterset_class is not None:
+                        pk_list = [obj.pk for obj in self.filterset_class(request.GET, model.objects.only("pk")).qs]
+                    else:
+                        pk_list = model.objects.values_list("pk", flat=True)
+                else:
+                    pk_list = request.POST.getlist("pk")
+                    initial = {
+                        "pk": pk_list,
+                        "return_url": view.get_return_url(request),
+                    }
+                    form = form_class(initial=initial)
+                table = view.table_class(view.queryset.filter(pk__in=pk_list), orderable=False)
+            elif view.action == "bulk_create":
+                form = view.get_form()
+            elif view.action == "bulk_update":
+                if request.POST.get("_all") and view.filterset_class is not None:
+                    pk_list = [obj.pk for obj in view.filterset_class(request.GET, view.queryset.only("pk")).qs]
```

Comment:
Something seems wrong here - accessing both `request.POST` and `request.GET`?

Reply:
It is the same behavior in `generic.BulkDeleteView.post()`. I will look at what the two methods contain.

Reply:
Yeah it should be request.POST